catalog/core/catalog-core-solrcommands/src/test/java/org/codice/ddf/commands/solr/RestoreCommandTest.java

```diff
@@ -0,0 +1,714 @@
+    when(mockSolrClient.optimize(eq(collection))).thenReturn(optimizationResponse);
+
+    NamedList<Object> responseHeader = getResponseHeader(restoreStatusCode);
+
+    NamedList<Object> mockResponse = new NamedList<>();
+    mockResponse.add("responseHeader", responseHeader);
+    if (restoreErrorMessages != null) {
+      mockResponse.add("failure", restoreErrorMessages);
+    } else {
+      mockResponse.add("success", new Object());
+    }
+
+    if (collection != null) {
+      when(mockSolrClient.request(any(SolrRequest.class), eq(collection))).thenReturn(mockResponse);
+    }
+  }
+
+  private void setupMockSolrClientForCollectionOptimization(
+      String collection, int optimizationStatusCode) throws Exception {
+    UpdateResponse optimizationResponse = getMockOptimizationResponse(optimizationStatusCode);
+    when(mockSolrClient.optimize(eq(collection))).thenReturn(optimizationResponse);
+  }
+
+  private void setupMockSolrClientForCollectionOptimizationThrowsException(String collection)
+      throws Exception {
+    when(mockSolrClient.optimize(eq(collection))).thenThrow(SolrServerException.class);
+  }
+
+  private void setupMockSolrClientForRestoreStatusFailure(NamedList<String> restoreErrorMessages)
+      throws Exception {
+    NamedList<Object> response =
+        getResponseForStatus(FAILURE_STATUS_CODE, RequestStatusState.FAILED, restoreErrorMessages);
+    when(mockSolrClient.request(any(SolrRequest.class), isNull(String.class))).thenReturn(response);
+  }
+
+  private void setupMockSolrClientForStatusThrowsException() throws Exception {
+    when(mockSolrClient.request(any(SolrRequest.class), isNull(String.class)))
+        .thenThrow(SolrServerException.class);
+  }
+
+  /**
+   * See
+   * https://cwiki.apache.org/confluence/display/solr/Collections+API#CollectionsAPI-BACKUP:BackupCollection
+   * for requests and responses.
+   */
+  private NamedList<Object> getResponseForStatus(
+      int statusCode, RequestStatusState requestStatusState, NamedList<String> errorMessages) {
+    NamedList<Object> responseHeader = getResponseHeader(statusCode);
+    NamedList<String> status = getStatus(requestStatusState);
+    NamedList<Object> response = new NamedList<>();
+    response.add("status", status);
+    response.add("responseHeader", responseHeader);
+    if (errorMessages != null) {
+      response.add("failure", errorMessages);
+    } else {
+      response.add("success", new Object());
+    }
+
+    return response;
+  }
+
+  /**
+   * On https://github.com/apache/lucene-solr see
+   * CollectionAdminRequest.RequestStatusResponse.getRequestStatus() for example usage.
+   * https://github.com/apache/lucene-solr/blob/master/solr/solrj/src/java/org/apache/solr/client/solrj/request/CollectionAdminRequest.java#L1343-L1346
+   */
+  private NamedList<String> getStatus(RequestStatusState requestStatusState) {
+    NamedList<String> status = new NamedList();
+    if (requestStatusState == RequestStatusState.FAILED) {
```

Comment:
Wouldn't `status.add("state", requestStatusState.getKey())` work?

Reply:
There was actually other duplicated code that I pulled into SolrCommandTest.